test/distributed/fsdp/test_distributed_checkpoint.py

```diff
@@ -47,6 +47,7 @@ def world_size(self):
         return 2
 
     @skip_if_lt_x_gpu(2)
+    @with_temp_dir
```

Comment:
pytest has a tempdir fixture if I'm not mistaken, but this works too.

Reply:
This is a distributed version with_temp_dir which will broadcast the path from rank0 to all other ranks.